engine/flink/avro-util/src/main/scala/pl/touk/nussknacker/engine/avro/schemaregistry/confluent/formatter/ConfluentAvroMessageReader.scala

```diff
@@ -54,15 +54,27 @@ private[confluent] class ConfluentAvroMessageReader(schemaRegistryClient: Schema
         } else {
           val keyString = str.substring(0, keyIndex)
           val valueString = if (keyIndex + 1 > str.length) "" else str.substring(keyIndex + 1)
-          val serializedKey = if (useStringForKey) {
-            keyString.getBytes(StandardCharsets.UTF_8)
-          } else {
-            val key = jsonToAvro(keyString, keySchema)
-            serializeImpl(keySubject, key, new AvroSchema(keySchema))
+          if (keySchema != null && keyString.length == 0) {
+            // key schema is provided but there is no content
+            throw new SerializationException("No key content found in line " + str)
+          } else if (keySchema == null) {
```

Comment:
Can you change `null` to  `Option`?

Reply:
This is temporary change to keep coherence with unchanged `readSchemaId`. It is removed in next PR that improves test data formatter.

Reply:
Changed null to Option.